src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -204,6 +250,78 @@ impl NamespacedKubernetesOrchestrator {
 
 #[async_trait]
 impl NamespacedOrchestrator for NamespacedKubernetesOrchestrator {
+    async fn service_metrics(&self, id: &str) -> Result<Vec<ServiceProcessMetrics>, anyhow::Error> {
+        let StatefulSet { spec: Some(StatefulSetSpec { replicas: Some(scale), ..}), .. } = self.stateful_set_api.get(id).await? else {
```

Comment:
This seems weird! The orchestrator is the authority on what the scale should be. Do we just need to keep track of that in memory, so we don't need to go asking Kubernetes about it?

Reply:
Yeah, that's a good point